src/helpers/buffers/ordering_mpsc.rs

```diff
@@ -194,6 +197,37 @@ impl<M: Message> OrderingMpscReceiver<M> {
     }
 }
 
+impl<M: Message> Debug for OrderingMpscReceiver<M> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        if cfg!(debug_assertions) {
+            write!(f, "OrderingMpscReceiver[{}]", self.name)
+        } else {
+            write!(f, "OrderingMpscReceiver")
+        }
+    }
+}
+
+/// [`OrderingMpscReceiver`] is a [`Stream`] that yields chunks of maximum capacity
+/// this instance can hold.
+impl<M: Message> Stream for OrderingMpscReceiver<M> {
+    type Item = Vec<u8>;
+
+    fn poll_next(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
+        let this = Pin::get_mut(self);
+        loop {
```

Comment:
I think that this is correct, though the (renamed) async take_next() is quite a bit more complex.  I think that it has to be, so that's OK, but it is maybe worth highlighting.

The subtle difference is that that code can't block if there are available unread items in the channel, whereas this never blocks, so it doesn't need to worry about that.  It makes be wonder whether we can't just rely on the provided implementation of next() that the stream provides.

Reply:
The reason I left `take_next` was because it gives extra functionality (take N items) whereas stream must always return chunks of the same size. Given that `take_next` is quite well tested I couldn't force myself to remove it because we may need it one day maybe.

Reply:
> stream must always return chunks of the same size

I don't think that is true.  We're returning a vec, which could be any size.

Reply:
That's true, the type system does not have this limitation, the implementation however does exactly that